fix(jans-fido2) : AppleAttestationProcessorTest fails (5/7) — mocks reference removed getRootCertificatesBySubjectDN instead of getAppleRootCertificates

Describe the bug

io.jans.fido2.service.processor.attestation.AppleAttestationProcessorTest fails with 5 failures out of 7 tests during the jans-fido2 build (observed in Jenkins/CI):


[ERROR] Tests run: 7, Failures: 5, Errors: 0, Skipped: 0
[ERROR] AppleAttestationProcessorTest.process_validData_success <<< FAILURE!
The test is out of sync with the production code. PR #13737 ("make Apple WebAuthn root CA subject DN configurable", merged 2026-04-24) changed AppleAttestationProcessor Step 1 to call attestationCertificateService.getAppleRootCertificates(), but the test (last updated 2024-12-12) still stubs and verifies the old getRootCertificatesBySubjectDN(anyString()).

Because Mockito mocks do not delegate to the real wrapper, the unstubbed getAppleRootCertificates() returns null, and AppleAttestationProcessor.java:112 (trustAnchorCertificates.size()) throws a NullPointerException. This fails exactly the 5 tests that reach root-certificate verification (Step 1); the 2 that return earlier (getAttestationFormat_valid_apple, process_ifCertificatesIsEmpty_badRequestException) pass.

To Reproduce
Steps to reproduce the behavior:

Check out main (or any branch since PR #13737).
Build the FIDO2 server module: mvn -pl server test -Dtest=AppleAttestationProcessorTest (under jans-fido2/).
Observe 5 of 7 tests fail with assertion failures stemming from a NullPointerException at AppleAttestationProcessor.java:112.
Expected behavior
All 7 tests in AppleAttestationProcessorTest pass. The mocks/verifications should target getAppleRootCertificates() — the method the production code actually invokes.

Screenshots
N/A — failure is in the CI/console test output (snippet above).

Desktop (please complete the following information):

OS: N/A (CI/build issue — reproducible on Linux/Jenkins and Windows)
Browser: N/A
Version: jans-fido2 0.0.0-nightly
Smartphone (please complete the following information):

Device: N/A
OS: N/A
Browser: N/A
Version: N/A
Additional context

Affected file (source): jans-fido2/server/src/main/java/io/jans/fido2/service/processor/attestation/AppleAttestationProcessor.java (line 111 calls getAppleRootCertificates()).
Affected file (test): jans-fido2/server/src/test/java/io/jans/fido2/service/processor/attestation/AppleAttestationProcessorTest.java.
Root cause: introduced by PR #13737; test not updated alongside the source change.
Fix: replace all stubs/verifications of getRootCertificatesBySubjectDN(anyString()) with getAppleRootCertificates() (10 occurrences across the 5 affected tests).

--- a/jans-fido2/server/src/test/java/io/jans/fido2/service/processor/attestation/AppleAttestationProcessorTest.java
+++ b/jans-fido2/server/src/test/java/io/jans/fido2/service/processor/attestation/AppleAttestationProcessorTest.java
@@ -124,7 +124,7 @@ class AppleAttestationProcessorTest {
         when(x5cNode.elements()).thenReturn(Collections.singletonList((JsonNode) new TextNode("x5c item")).iterator());
         X509Certificate credCert = mock(X509Certificate.class);
         when(certificateService.getCertificate(anyString())).thenReturn(credCert);
-        when(attestationCertificateService.getRootCertificatesBySubjectDN(anyString())).thenThrow(new Fido2RuntimeException("test exception"));
+        when(attestationCertificateService.getAppleRootCertificates()).thenThrow(new Fido2RuntimeException("test exception"));
         when(credCert.getIssuerDN()).thenReturn(new BasicUserPrincipal("test issuer dn"));
         when(errorResponseFactory.badRequestException(any(), anyString())).thenThrow(new WebApplicationException(Response.status(400).entity("test exception").build()));
 
@@ -136,7 +136,7 @@ class AppleAttestationProcessorTest {
 
         verify(log).info(eq("AttStmt: test_att_stmt"));
         verify(certificateService).getCertificate(eq("x5c item"));
-        verify(attestationCertificateService).getRootCertificatesBySubjectDN(anyString());
+        verify(attestationCertificateService).getAppleRootCertificates();
         verify(log).warn(eq("Failed to find attestation validation signature public certificate with DN: '{}'"), eq("test issuer dn"));
         verify(errorResponseFactory).badRequestException(any(), eq("Failed to find attestation validation signature public certificate with DN: test issuer dn"));
         verifyNoMoreInteractions(log, errorResponseFactory);
@@ -159,7 +159,7 @@ class AppleAttestationProcessorTest {
         X509Certificate credCert = mock(X509Certificate.class);
         when(certificateService.getCertificate(anyString())).thenReturn(credCert);
         List<X509Certificate> rootCertificates = Collections.singletonList(mock(X509Certificate.class));
-        when(attestationCertificateService.getRootCertificatesBySubjectDN(anyString())).thenReturn(rootCertificates);
+        when(attestationCertificateService.getAppleRootCertificates()).thenReturn(rootCertificates);
         when(certificateVerifier.verifyAttestationCertificates(anyList(), anyList())).thenReturn(mock(X509Certificate.class));
         when(authData.getAuthDataDecoded()).thenReturn("test decoded".getBytes());
         when(commonUtilService.writeOutputStreamByteList(anyList())).thenThrow(new IOException("test ioexception"));
@@ -173,7 +173,7 @@ class AppleAttestationProcessorTest {
 
         verify(log).info(eq("AttStmt: test_att_stmt"));
         verify(certificateService).getCertificate(eq("x5c item"));
-        verify(attestationCertificateService).getRootCertificatesBySubjectDN(anyString());
+        verify(attestationCertificateService).getAppleRootCertificates();
         verify(log).debug(eq("APPLE_WEBAUTHN_ROOT_CA root certificate: 1"));
         verify(certificateVerifier).verifyAttestationCertificates(anyList(), anyList());
         verify(log).info(eq("Step 1 completed"));
@@ -199,7 +199,7 @@ class AppleAttestationProcessorTest {
         X509Certificate credCert = mock(X509Certificate.class);
         when(certificateService.getCertificate(anyString())).thenReturn(credCert);
         List<X509Certificate> rootCertificates = Collections.singletonList(mock(X509Certificate.class));
-        when(attestationCertificateService.getRootCertificatesBySubjectDN(anyString())).thenReturn(rootCertificates);
+        when(attestationCertificateService.getAppleRootCertificates()).thenReturn(rootCertificates);
         when(certificateVerifier.verifyAttestationCertificates(anyList(), anyList())).thenReturn(mock(X509Certificate.class));
         when(authData.getAuthDataDecoded()).thenReturn("test decoded".getBytes());
         ByteArrayOutputStream baos = mock(ByteArrayOutputStream.class);
@@ -216,7 +216,7 @@ class AppleAttestationProcessorTest {
 
         verify(log).info(eq("AttStmt: test_att_stmt"));
         verify(certificateService).getCertificate(eq("x5c item"));
-        verify(attestationCertificateService).getRootCertificatesBySubjectDN(anyString());
+        verify(attestationCertificateService).getAppleRootCertificates();
         verify(log).debug(eq("APPLE_WEBAUTHN_ROOT_CA root certificate: 1"));
         verify(certificateVerifier).verifyAttestationCertificates(anyList(), anyList());
         verify(log).info(eq("Step 1 completed"));
@@ -245,7 +245,7 @@ class AppleAttestationProcessorTest {
         X509Certificate credCert = mock(X509Certificate.class);
         when(certificateService.getCertificate(anyString())).thenReturn(credCert);
         List<X509Certificate> rootCertificates = Collections.singletonList(mock(X509Certificate.class));
-        when(attestationCertificateService.getRootCertificatesBySubjectDN(anyString())).thenReturn(rootCertificates);
+        when(attestationCertificateService.getAppleRootCertificates()).thenReturn(rootCertificates);
         when(certificateVerifier.verifyAttestationCertificates(anyList(), anyList())).thenReturn(mock(X509Certificate.class));
         when(authData.getAuthDataDecoded()).thenReturn("test decoded".getBytes());
         ByteArrayOutputStream baos = mock(ByteArrayOutputStream.class);
@@ -264,7 +264,7 @@ class AppleAttestationProcessorTest {
 
         verify(log).info(eq("AttStmt: test_att_stmt"));
         verify(certificateService).getCertificate(eq("x5c item"));
-        verify(attestationCertificateService).getRootCertificatesBySubjectDN(anyString());
+        verify(attestationCertificateService).getAppleRootCertificates();
         verify(log).debug(eq("APPLE_WEBAUTHN_ROOT_CA root certificate: 1"));
         verify(certificateVerifier).verifyAttestationCertificates(anyList(), anyList());
         verify(log).info(eq("Step 1 completed"));
@@ -295,7 +295,7 @@ class AppleAttestationProcessorTest {
         X509Certificate credCert = mock(X509Certificate.class);
         when(certificateService.getCertificate(anyString())).thenReturn(credCert);
         List<X509Certificate> rootCertificates = Collections.singletonList(mock(X509Certificate.class));
-        when(attestationCertificateService.getRootCertificatesBySubjectDN(anyString())).thenReturn(rootCertificates);
+        when(attestationCertificateService.getAppleRootCertificates()).thenReturn(rootCertificates);
         when(certificateVerifier.verifyAttestationCertificates(anyList(), anyList())).thenReturn(mock(X509Certificate.class));
         when(authData.getAuthDataDecoded()).thenReturn("test decoded".getBytes());
         ByteArrayOutputStream baos = mock(ByteArrayOutputStream.class);
@@ -317,7 +317,7 @@ class AppleAttestationProcessorTest {
 
         verify(log).info(eq("AttStmt: test_att_stmt"));
         verify(certificateService).getCertificate(eq("x5c item"));
-        verify(attestationCertificateService).getRootCertificatesBySubjectDN(anyString());
+        verify(attestationCertificateService).getAppleRootCertificates();
         verify(log).debug(eq("APPLE_WEBAUTHN_ROOT_CA root certificate: 1"));
         verify(certificateVerifier).verifyAttestationCertificates(anyList(), anyList());
         verify(log).info(eq("Step 1 completed"));

